node/packages/aws-lambda-sdk/instrument/lib/auto-send-spans.js

```diff
@@ -38,6 +38,11 @@ const sendData = () => {
   };
   pendingSpans.length = 0;
   pendingCapturedEvents.length = 0;
+  if (!invocationContextAccessor.value) {
+    // Root span comes with "aws.lambda.*" tags, which require unconditionally requestId
+    // which we don't have if handler crashed at initialization.
+    payload.spans = payload.spans.filter((span) => span.name !== 'aws.lambda');
```

Comment:
Does this mean the root span will never get propagated when there is an initialization error or does this mean that it will be held until we receive a requestId? 🤔

Reply:
It will never be propagated, as lambda is restarted at this point (we will never now the request id)

Reply:
Ahh okay got it, this makes sense. Thanks for the clarification 😎

Reply:
We probably need to discuss this request-id requirement, for now it's best we can do

Reply:
Yeah for sure. I am going through some code clean up on the dev mode extension right now and I am trying to think through some options where we don't have to rely on it at all and can just rely on the `traceId`.

I think it is possible I just need to do a few experiments to test a few theories. Maybe we can set up a time to discuss next week sometime?

Reply:
Yes, discussing it next week is a good idea